src/Resource/AttributeResource.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Router\Resource;
+
+use olvlvl\ComposerAttributeCollector\Attributes;
+use Yiisoft\Router\Group;
+use Yiisoft\Router\Route;
+
+/**
+ * An attribute resource represents routes that declared via PHP Attributes.
+ * @codeCoverageIgnore
+ */
+final class AttributeResource implements ResourceInterface
+{
+    /**
+     * @var array<class-string, \ReflectionMethod>
+     */
+    private static array $reflectionsCache = [];
+
+    public function getRoutes(): array
+    {
+        $routes = [];
+        $groupRoutes = [];
+        $routePredicate = Attributes::predicateForAttributeInstanceOf(Route::class);
+        $targetMethods = Attributes::filterTargetMethods($routePredicate);
+        foreach ($targetMethods as $targetMethod) {
```

Comment:
Is it parse all PHP files? Can we search in a specific folder?

Reply:
Yes, it's a composer plugin, can be configured in composer.json. We may change this implementation in the future (perhaps when `yiisoft/classifier` is ready).